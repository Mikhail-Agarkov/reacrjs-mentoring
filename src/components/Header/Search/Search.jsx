import React from 'react';
import Filter from '../../Filter/Filter';
import styles from './search.module.scss'

export default class Search extends React.PureComponent {

    constructor(props) {
        super(props);
        this.searchByTitle = 'title';
        this.searchByGenre = 'genre';
        this.state = {
            searchBy: this.searchByTitle
        };
        this.input = React.createRef();
    }

    searchHandler = (event) => {
        event.preventDefault();
        //api call (this.input.current.value, this.state.searchBy)
    };

    searchByHandler = (type) => {
        this.setState({searchBy: type});
    };

    render() {
        return (
            <section className={styles.content}>
                <h2 className={styles.title}>{this.props.title}</h2>
                <form className={styles.form} onSubmit={this.searchHandler}>
                    <input type='text'  className={styles.search} placeholder={this.props.placeholder} ref={this.input}/>
                    <button className={styles.button}>{this.props.button}</button>
                </form>
                <Filter title='Search by' buttons={
                    [
                        {text: this.searchByTitle, handler: () => this.searchByHandler(this.searchByTitle), key: 1},
                        {text: this.searchByGenre, handler: () => this.searchByHandler(this.searchByGenre), key: 2}
                    ]
                }/>
            </section>
        );
    }
}