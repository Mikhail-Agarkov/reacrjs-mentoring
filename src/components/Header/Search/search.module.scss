.body {
  max-width: 1000px;
  margin: 100px auto;
}

.search {
  border: none;
  outline: none;
  height: 57px;
  width: 713px;
  opacity: 0.7;
  color: #fff;
  border-radius: 4px;
  background-color: rgba(50, 50, 50, 0.8);
  font-size: 20px;
  padding-left: 20px;
  font-family: Montserrat, sans-serif;
}

.form {
  display: flex;
  justify-content: space-between;
}

.button {
  border: none;
  outline: none;
  height: 57px;
  width: 233px;
  border-radius: 4px;
  background-color: #F65261;
  color: #FFFFFF;
  font-family: Montserrat, sans-serif;
  font-size: 20px;
  font-weight: 700;
  text-transform: uppercase;
  cursor: pointer;

  &:hover {
    background-color: #dd4957;
  }

  &:active {
    background-color: #c4414d;
  }
}

.title {
  height: 49px;
  width: 393px;
  color: #FFFFFF;
  font-family: Montserrat, sans-serif;
  font-size: 40px;
  font-weight: 300;
  letter-spacing: 1px;
  line-height: 49px;
  text-align: center;
  text-transform: uppercase;
}