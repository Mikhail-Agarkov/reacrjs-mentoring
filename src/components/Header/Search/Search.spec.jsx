import React from 'react';
import {shallow, mount} from 'enzyme';

import Search from "./Search";

describe('Search', () => {
    it('should render Search correctly', () => {
        const component = shallow(<Search title='Find your movie' placeholder='Search' button='Search'/>);
        expect(component).toMatchSnapshot();
    });

    it('should perform search on button click', () => {
        const component = shallow(<Search title='Find your movie' placeholder='Search' button='Search'/>);
        component.find('form').simulate('submit');
        //expect smthing when API call is ready
    });

    it('should change "searchBy" state to "title"', () => {
        const component = mount(<Search title='Find your movie' placeholder='Search' button='Search'/>);
        component.find('Filter div button').first().simulate('click');
        expect(component.state('searchBy')).toEqual('title');
    });

    it('should change "searchBy" state to "genre"', () => {
        const component = mount(<Search title='Find your movie' placeholder='Search' button='Search'/>);
        component.find('Filter div button').last().simulate('click');
        expect(component.state('searchBy')).toEqual('genre');
    });
});