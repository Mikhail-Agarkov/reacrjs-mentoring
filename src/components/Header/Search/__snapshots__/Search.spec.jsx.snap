// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render with given state from Redux store 1`] = `
<section>
  <h2
    className="title"
  >
    Find your movie
  </h2>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      className="search"
      placeholder="Search"
      type="text"
    />
    <button
      className="button"
    >
      Search
    </button>
  </form>
  <div
    className="content"
  >
    <span
      className="text"
    >
      Search by
    </span>
    <div
      className="buttonGroup"
    >
      <button
        className="button"
        data-selected={true}
        onClick={[Function]}
      >
        Title
      </button>
      <button
        className="button"
        data-selected={false}
        onClick={[Function]}
      >
        Genre
      </button>
    </div>
  </div>
</section>
`;
